src/coreclr/vm/callconvbuilder.cpp

```diff
@@ -11,9 +11,12 @@
 
 // According to ECMA-335, type name strings are UTF-8. Since we are
 // looking for type names that are equivalent in ASCII and UTF-8,
-// using a const char constant is acceptable. Type name strings are
-// in Fully Qualified form, so we include the ',' delimiter.
-#define MAKE_FULLY_QUALIFIED_CALLCONV_TYPE_NAME_PREFIX(callConvTypeName) CMOD_CALLCONV_NAMESPACE "." callConvTypeName ","
+// using a const char constant is acceptable.
+#define MAKE_FULLY_QUALIFIED_CALLCONV_TYPE_NAME(callConvTypeName) CMOD_CALLCONV_NAMESPACE "." callConvTypeName
+
+// We can include the ',' delimiter, when a type name is in Fully Qualified
+// form outside its declaring assembly - most common scenario.
+#define MAKE_FULLY_QUALIFIED_CALLCONV_TYPE_NAME_PREFIX(callConvTypeName) MAKE_FULLY_QUALIFIED_CALLCONV_TYPE_NAME(callConvTypeName) ","
```

Comment:
Instead of duplicating the type names once without `,` and second time with `,`, would it better to introduce `EqualsSimpleTypeName` matching method that succeeds when we have end-of-string or comma after the match?

Reply:
I was curious if someone was going to balk at the duplicate string arrangement. I can reduce it for sure.